lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -414,16 +414,40 @@ SPIRVType *LLVMToSPIRV::transType(Type *T) {
     if (Name == getSPIRVTypeName(kSPIRVTypeName::ConstantPipeStorage))
       return transType(getPipeStorageType(M));
 
-    auto *Struct = BM->openStructType(T->getStructNumElements(), Name.str());
+    const size_t MaxNumElements = 65535 - 2;
+    const size_t NumElements = T->getStructNumElements();
+    size_t SPIRVStructNumElements = NumElements;
+    if (NumElements > MaxNumElements &&
+        BM->isAllowedToUseExtension(
+            ExtensionID::SPV_INTEL_long_constant_composite)) {
+      SPIRVStructNumElements = MaxNumElements;
+    }
```

Comment:
What happens if `NumElements > MaxNumElements`, but extension is not enabled? Shall we emit an error in that case?

Reply:
It will be emitted through `validate` method of `SPIRVTypeStruct` class.